libtiledbsoma/src/cli/cli.cc

```diff
@@ -33,6 +33,8 @@
 #include "soma/enums.h"
 #include "soma/soma_array.h"
 #include "utils/arrow_adapter.h"
+#include "utils/carrow.h"
+#include "utils/logger.h"
```

Comment:
Why were these headers added?

Reply:
`logger.h` is used for the `LOG_*` macros and `fmt`. `carrow.h` is used for `ArrowSchema`. `soma_array.h` use to include `logger.h` but that has now been removed.

Reply:
But `soma_array.h` includes `logger_public.h`. Isn't it enough?

Reply:
No, `logger_public.h` does not expose `fmt`

Reply:
FWIW one can (somewhat poorly) drop in `tinyfmt` which is a less powerful than `fmt` but a single-header C++11 library.  The key is, as @nguyenv points out, that the public logger only takes a string (or `const char*`, I forgot) to 'report' and does no formatting whatsoever itself.